src/main/kotlin/com/app/ui/screens/games/SudokuScreen.kt

```diff
@@ -0,0 +1,200 @@
+        ) {
+            SudokuBoardUI(
+                sudokuBoard = board,
+                selectedNode = selectedCell,
+                completed = completed,
+                onNodeClick = { x, y ->
+                    selectedCell = Pair(x, y)
+                }
+            )
+
+            Spacer(modifier = Modifier.width(16.dp))
+
+            NumberPad(
+                onNumberClick = { number ->
+                    selectedNumber = number
+                    selectedCell?.let { (x, y) ->
+                        if (selectedNumber != null) {
+                            scope.launch {
+                                val updatedBoard = updateCell(board, x, y, selectedNumber!!)
+                                board = updatedBoard.first
+                                completed = updatedBoard.second
+                            }
+                        }
+                    }
+                }
+            )
+        }
+
+        Row(
+            modifier = Modifier
+                .padding(4.dp),
+            horizontalArrangement = Arrangement.SpaceEvenly
+        ) {
+            Button(
+                onClick = onBack,
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = Color(0xFFC5705D),
+                    contentColor = Color.White
+                )
+            ) {
+                Text("Main Menu")
+            }
+
+            Spacer(modifier = Modifier.width(8.dp))
+
+            // Saving current state of the board to the database
+            Button(
+                onClick = {
+                    scope.launch {
+                        val temp = board
+                        board = null
+                        board = saveSudoku(temp)
+                    }
+                },
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = Color(0xFFC5705D),
+                    contentColor = Color.White
+                )
+            ) {
+                Text("Save progress")
+            }
+
+            Spacer(modifier = Modifier.width(8.dp))
+
+            // Displaying the basic board to the user.
+            Button(
+                onClick = {
+                    scope.launch {
+                        try {
+                            board = null
+                            board = newGame()
+                        } catch (e: Exception) {
+                            println("Error during creating new game: ${e.message}")
+                        }
+                    }
+                },
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = Color(0xFFC5705D),
+                    contentColor = Color.White
+                )
+            ) {
+                Text("New game")
+            }
+        }
+    }
+}
+
+// Updating board state
+fun updateCell(board: SudokuBoard?, x: Int, y: Int, number: Int): Pair<SudokuBoard, Boolean> {
```

Comment:
```suggestion
fun updateCell(board: SudokuBoard, x: Int, y: Int, number: Int): Pair<SudokuBoard, Boolean> {
```

i tak nie obsługuje Pan tutaj przypadku, że board jest nullem